rust/src/dns/detect.rs

```diff
@@ -114,9 +116,44 @@ pub unsafe extern "C" fn rs_dns_detect_opcode_free(ptr: *mut c_void) {
     }
 }
 
+/// Perform the DNS rrtype match.
+///
+/// 1 will be returned on match, otherwise 0 will be returned.
+#[no_mangle]
+pub extern "C" fn rs_dns_rrtype_match(
+    tx: &mut DNSTransaction, detect: &mut DetectUintData<u16>, rrtype: u16,
```

Comment:
Seeing this failure: https://github.com/OISF/suricata/actions/runs/7726431734/job/21062732402?pr=10289#step:13:434

I think that this last parameter here should be `flags: u8`, as we see for opcode. The info for rrtype should be present in the `detect` param, from what I understand 🤔